dali/kernels/erase/erase_gpu.h

```diff
@@ -503,24 +505,22 @@ struct EraseGpu {
       auto &sample = sample_desc_cpu[i];
       sample.in = in.data[i];
       sample.out = out.data[i];
-      sample.erase_regions = erased_regions.data[i];
+      auto *regions_ptr = ctx.scratchpad->ToGPU(stream, erased_regions[i]);
+      sample.erase_regions = make_span(regions_ptr, erased_regions[i].size());
```

Comment:
Aren't you doing `num_samples` memcopies to device here? You can do just one, and still point to specific offset corresponding to erased_region of given sample.